File: apps/desktop/src-tauri/src/commands/library.rs
use tauri::State;

use crate::{
    errors::{CommandResult, IntoCommandResult},
    models::library::{ExplorerSnapshot, LibrarySnapshot},
    services::LibraryService,
};

#[tauri::command]
pub async fn load_library_snapshot(
    library_service: State<'_, LibraryService>,
) -> CommandResult<LibrarySnapshot> {
    library_service.load_library_snapshot().await.into_command()
}

#[tauri::command]
pub async fn load_explorer_snapshot(
    library_service: State<'_, LibraryService>,
) -> CommandResult<ExplorerSnapshot> {
    library_service.load_explorer_snapshot().await.into_command()
}

File: apps/desktop/src/shared/lib/ipc/library.ts
import type { ExplorerSnapshot, LibrarySnapshot } from '../../types';
import { invokeCamel } from './core';

export const libraryIpc = {
  loadLibrarySnapshot: (): Promise<LibrarySnapshot> => invokeCamel('load_library_snapshot'),
  loadExplorerSnapshot: (): Promise<ExplorerSnapshot> => invokeCamel('load_explorer_snapshot'),
};
